fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/RdfContext.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.api;
+
+/**
+ * A collection of RDF contexts that can be used to extract triples from FedoraResources
+ *
+ * @author acoburn
+ * @since Dec 4, 2015
+ */
+public enum RdfContext {
```

Comment:
Can I ask the question another way? I don't grok how the enum part has
anything to do with the stream/iterator part; that seems like it should
have been another ticket, and another PR. I trust @acoburn's assessment
about whether that's possible or not, but this seems like a PR that is
trying to to do a little too much. Maybe that's a consequence of mvn
juggling?

On Wed, Feb 17, 2016 at 10:47 AM, Andrew Woods <notifications@github.com>
wrote:

> In fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/RdfContext.java
> <https://github.com/fcrepo4/fcrepo4/pull/962#discussion_r53182947>:
>
> > + *
> > + * Unless required by applicable law or agreed to in writing, software
> > + * distributed under the License is distributed on an "AS IS" BASIS,
> > + * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > + * See the License for the specific language governing permissions and
> > + * limitations under the License.
> > + */
> > +package org.fcrepo.kernel.api;
> > +
> > +/**
> > + * A collection of RDF contexts that can be used to extract triples from FedoraResources
> > + *
> > + * @author acoburn
> > + * @since Dec 4, 2015
> > + */
> > +public enum RdfContext {
>
> Agreed. My preference is for incremental steps. @acoburn
> <https://github.com/acoburn> has done a great job of initial decoupling
> from fcrepo-kernel-modeshape with this PR. I would be inclined to move
> forward with the enum approach for this increment.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fcrepo4/fcrepo4/pull/962/files#r53182947>.
>